zetaclient/btc_signer_test.go

```diff
@@ -253,27 +253,27 @@ func TestSelectUTXOs(t *testing.T) {
 	require.Equal(t, 0.01, amount)
 	require.Equal(t, ob.utxos[0:1], result)
 
-	// // Case2: nonce = 1, must FAIL and wait for previous transaction to be mined
-	// // 		input: utxoCap = 5, amount = 0.5, nonce = 1
-	// // 		output: error
-	// result, amount, err = ob.SelectUTXOs(0.5, 5, 1, tssAddress)
-	// require.NotNil(t, err)
-	// require.Nil(t, result)
-	// require.Equal(t, 0.0, amount)
-	// require.Equal(t, "findNonceMarkUTXO: transaction 0-mgaRVNhouhVaiKx8xVtLNHBbSUe1o36qZJ-0 not mined yet", err.Error())
-	// mineTxNSetNonceMark(ob, 0, dummyTxID, -1) // mine a transaction for nonce 0
-
-	// // Case3: nonce = 1, must FAIL without nonce mark utxo
-	// // 		input: utxoCap = 5, amount = 0.5, nonce = 1
-	// // 		output: error
-	// result, amount, err = ob.SelectUTXOs(0.5, 5, 1, tssAddress)
-	// require.NotNil(t, err)
-	// require.Nil(t, result)
-	// require.Equal(t, 0.0, amount)
-	// require.Equal(t, "findNonceMarkUTXO: cannot find nonce-mark utxo with nonce 0", err.Error())
+	// Case2: nonce = 1, must FAIL and wait for previous transaction to be mined
+	// 		input: utxoCap = 5, amount = 0.5, nonce = 1
+	// 		output: error
+	result, amount, err = ob.SelectUTXOs(0.5, 5, 1, tssAddress)
+	require.NotNil(t, err)
+	require.Nil(t, result)
+	require.Equal(t, 0.0, amount)
+	require.Equal(t, "findNonceMarkUTXO: outTx 0-mgaRVNhouhVaiKx8xVtLNHBbSUe1o36qZJ-0 not included yet", err.Error())
```

Comment:
Maybe we can use `require.ErrorIs` here?

Reply:
Don't know how to make ErrorIs working in my test, instead of verifying a error message, it verifies whole error chain. I failed with below adjustment:
require.ErrorIs(t, err, errors.New("findNonceMarkUTXO: outTx 0-mgaRVNhouhVaiKx8xVtLNHBbSUe1o36qZJ-0 not included yet"))

Reply:
You can define a variable for the error and reuse the variable inside `ErrorIs`, maybe this is too much overhead in this case.
By experience it's sometimes harder to maintain when you write the error message to assert in the tests